docs/endpoint-hal.md

```diff
@@ -33,63 +40,49 @@ Here are the functions needed to be registered/unregistered inside `impl.c`:
 * unregister_device : unregistered device
 * DECLARE_DEVICE : this must be declared inside `impl.c`
 
-Add the new device into `hal/Makefile`:
```

Comment:
Why did you change the following?

Reply:
We won't use the `hal/Makefile` anymore.

Reply:
Only need the `impl.c` and `build.mk` in the platform.

Reply:
Then, you have to address such modifications in git commit message(s).